### PR TITLE
Add decode_options and hyp_cleaner in evaluate_whisper_inference

### PR DESCRIPTION
**Pull request by pyf98**
This PR adds `decode_options` and `hyp_cleaner` in `evaluate_whisper_inference`. The `decode_options` can be used to control the decoding hyperparameters in Whisper model's `transcribe` method.

Here is an example script to evaluate a test set using Whisper:

```bash
#!/usr/bin/env bash
# Set bash to 'debug' mode, it will exit on :
# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
set -e
set -u
set -o pipefail

whisper_tag=medium
cleaner=whisper_en
hyp_cleaner=whisper_en
nj=1
test_sets="test/WSJ/test_eval92"
decode_options="{language: en, task: transcribe, temperature: 0, beam_size: 10, fp16: False}"

for x in ${test_sets}; do
    wavscp=dump/raw/${x}/wav.scp
    outdir=whisper-${whisper_tag}_outputs/${x}
    gt_text=dump/raw/${x}/text

    scripts/utils/evaluate_asr.sh \
        --whisper_tag ${whisper_tag} \
        --nj ${nj} \
        --gpu_inference true \
        --stage 2 \
        --stop_stage 3 \
        --cleaner ${cleaner} \
        --hyp_cleaner ${hyp_cleaner} \
        --decode_options "${decode_options}" \
        --gt_text ${gt_text} \
        ${wavscp} \
        ${outdir}
done
```

### CODE REVIEW

File: egs2/TEMPLATE/asr1/pyscripts/utils/evaluate_whisper_inference.py [hunk 1]
```diff
@@ -88,7 +90,7 @@ def inference(
 
     # 3. Build data-iterator
     info_list = []
-    wavscp = open(data_path_and_name_and_type, "r", encoding="utf-8")
+    wavscp = open(key_file, "r", encoding="utf-8")
```
Comment (pyf98) [inline egs2/TEMPLATE/asr1/pyscripts/utils/evaluate_whisper_inference.py:93]: `key_file` can contain only a subset of utterances due to the use of multiple jobs. `data_path_and_name_and_type` can contain all the data.

File: egs2/TEMPLATE/asr1/scripts/utils/evaluate_asr.sh [hunk 2]
```diff
@@ -201,9 +205,11 @@ if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
 
     # 3. Concatenates the output files from each jobs
     for f in token token_int score text; do
-        for i in $(seq "${_nj}"); do
-            cat "${logdir}/output.${i}/1best_recog/${f}"
-        done | LC_ALL=C sort -k1 >"${outdir}/${f}"
+        if [ -f "${logdir}/output.1/1best_recog/${f}" ]; then
```
Comment (pyf98) [inline egs2/TEMPLATE/asr1/scripts/utils/evaluate_asr.sh:208]: Whisper outputs do not contain all the files.

**Comment by ftshijt** [comment]
Many thanks for the update! 

The examples are very good to show somewhere, do you consider some places for this? (A candidate might be tts/svs templates' readme as that is having a section for ASR evaluation specifically, but then it might not be general?)

File: doc/espnet2_tutorial.md [hunk 3]
```diff
@@ -254,6 +254,48 @@ You need to fill `model_name` by yourself. You can search for pretrained models
 
 (Deprecated: See the following link about our pretrain models: https://github.com/espnet/espnet_model_zoo)
 
+
+### Evaluation using OpenAI Whisper
```
Comment (pyf98) [inline doc/espnet2_tutorial.md:258]: Added this example in ESPnet2 doc, since it is a general script.

**Comment by sw005320** [comment]
Thanks, @pyf98!

**Pull request merged**